compiler/executable/pipeline.rs

```diff
@@ -182,7 +182,13 @@ pub(crate) fn compile_pipeline_stages(
         input_variables.enumerate().map(|(i, var)| (var, VariablePosition::new(i as u32))).collect();
 
     for stage in annotated_stages {
-        let selected_variables = selected_variables.iter().cloned().chain(stage.referenced_variables()).collect();
+        // TODO: We can filter out the variables that are no longer needed in the future stages, but are carried as selected variables from the previous one
```

Comment:
I'm scared of breaking too many existing things if I do so, and I just want to merge my stable fetch.